File: migrations/backend-api/app/models/pergunta_variavel.py
"""
Modelo Pivot PerguntaVariavel
"""
from sqlalchemy import Column, Integer, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class PerguntaVariavel(Base):
    __tablename__ = "pergunta_variavel"
    
    id = Column(Integer, primary_key=True, index=True)
    pergunta_id = Column(Integer, ForeignKey("perguntas.id"), nullable=False)
    variavel_id = Column(Integer, ForeignKey("variaveis.id"), nullable=False)
    
    # Timestamps
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    
    # Relationships
    pergunta = relationship("Pergunta", back_populates="pergunta_variaveis")
    variavel = relationship("Variavel", back_populates="pergunta_variaveis")

File: php_old_app/app/Models/PerguntaVariavel.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class PerguntaVariavel extends Model
{
    use HasFactory;

    protected $table = 'pergunta_variavel';

    protected $fillable = [
        'pergunta_id',
        'variavel_id',
    ];

    public function pergunta()
    {
        return $this->belongsTo(Pergunta::class);
    }

    public function variavel()
    {
        return $this->belongsTo(Variavel::class);
    }

}